docs/concepts/packages.md

```diff
@@ -3,111 +3,106 @@ id: packages
-
-The full list of pre-deployed available packages can be found under the [demo package](https://github.com/gnolang/gno/tree/master/examples/gno.land/p/demo). Below are some of the most commonly used packages.
-
-## `avl`
-
-In Go, the classic key/value data type is represented by the `map` construct. However, while Gno also supports the use of `map`, it is not a viable option as it lacks determinism due to its non-sequential nature.
- 
-To address this issue, Gno implements the [AVL Tree](https://en.wikipedia.org/wiki/AVL\_tree) (Adelson-Velsky-Landis Tree) as a solution. The AVL Tree is a self-balancing binary search tree.
-
-The `avl` package comprises a set of functions that can manipulate the leaves and nodes of the AVL Tree.
-
-## `grc20`
-
-Gno includes an implementation of the `erc20` fungible token standard referred to as `grc20`. The interfaces of `grc20` are as follows:
-
-[embedmd]:# (../assets/explanation/packages/pkg-1.gno go)
 ```go
-func TotalSupply() uint64
-func BalanceOf(account std.Address) uint64
-func Transfer(to std.Address, amount uint64)
-func Approve(spender std.Address, amount uint64)
-func TransferFrom(from, to std.Address, amount uint64)
-func Allowance(owner, spender std.Address) uint64
-```
-
-The role of each function is as follows:
+package myrealm
 
-* `TotalSupply`: Returns the total supply of the token.
-* `BalanceOf`: Returns the balance of tokens of an account.
-* `Transfer`: Transfers specific `amount` of tokens from the `caller` of the function to the `to` address.
-* `Approve`: Grants the `spender`(also referred to as `operator`) with the ability to send specific `amount` of the `caller`'s (also referred to as `owner`) tokens on behalf of the `caller`.
-* `TransferFrom`: Can be called by the `operator` to send specific `amount` of `owner`'s tokens from the `owner`'s address to the `to` address.
-* `Allowance`: Returns the number of tokens approved to the `spender` by the `owner`.
+import (
+	"gno.land/p/demo/avl"
+)
 
-Two types of contracts exist in`grc20`:
+// This AVL tree will be persisted after transaction calls
+var tree *avl.Tree
 
-1. `AdminToken`\
-   \- Implements the token factory with `Helper` functions.\
-   \- The underlying struct should not be exposed to users. However, it can be typecasted as UserToken using the `GRC20()` method.
-2. `UserToken`\
-   \- Implements the `IGRC20` interface.\
-   \- The underlying struct can be exposed to users. Created with the `GRC20()` method of `adminToken`.
+func Set(key string, value int) {
+	// tree.Set takes in a string key, and an interface{} value
+	tree.Set(key, value)
+}
 
-## `grc721`
+func Get(key string) int {
+  // tree.Get returns the  interface{} value at given key, 
+  // and a bool to signify the existence of the key-value pair
+  rawValue, exists := tree.Get(key)
+  if !exists {
+	  panic("value at given key does not exist")
+  }
+  // rawValue needs to be converted into the proper type
+  value := rawValue.(int)
+  return value
```

Comment:
Nitpick: you can inline this

Reply:
[9fa722c](https://github.com/gnolang/gno/pull/1469/commits/9fa722c707e3ddcc21e49d41186e990388aa9f55)